### DISCUSSION TITLE
Fail2ban Oracle-Linux  firewalld  additional Actions - how make it right

### DISCUSSION THREAD

**Discussion by SystemSpam** [fail2ban/fail2ban#3814] [Q&A] [answered]
Hello.

I installed fail2ban on a Oracle-Linux with firewalld and it works like a charm.

I changed the banaction to

```
banaction = firewallcmd-rich-rules[actiontype=]
banaction_allports = firewallcmd-rich-rules[actiontype=]
```

and enabled the sshd section
```
[sshd]

enabled = true
port    = ssh
logpath = %(sshd_log)s
backend = %(sshd_backend)s
bantime = 1h
bantime.increment = true
bantime.factor = 24
bantime.maxtime = 5w
findtime = 2d
maxretry = 3
``` 
The richrules are set correctly for banning ip's

Now i want to push the ban to my ntfy.sh wit a simple curl -command.

I created a fie in /etc/fail2ban/actions.d  called "curlAction.conf" with a simple command.
```
[Definition]
actionstart =
actionstop =
actioncheck =
actionban = curl -d 'Ban IP : "<ip>"' -H "p: 3" -H "t: Oracle 1" ntfy.sh/<mynamespace>
actionunban =
```
In the jail.local I added the line 
`action = curlAction`
in the [sshd]-Section.

Now the push works, but the firewalld-rich-rules will not be set.

it looks like, the my "action = curlAction" disables the standard action ( %(action_)s ).
`action_ = %(banaction)s[port="%(port)s", protocol="%(protocol)s", chain="%(chain)s"]`

Similar to the other actions in the Jail.local (or jail.conf) the actions a combined 
```
action_mw = %(action_)s
            %(mta)s-whois[sender="%(sender)s", dest="%(destemail)s", protocol="%(protocol)s", chain="%(chain)s"]
```
I think that is my fault, but i didn't find any Information (or I'm to silly to search with the right search terms) how to do this.

I tried it as a multiline
```
actionban = %(action_)s
                     curl -d 'Ban IP : "<ip>"' -H "p: 3" -H "t: Oracle 1" ntfy.sh/<mynamespace> 
```
without any success.

Could anyone help me with this?

Thank You in advanced

**Comment by sebres** [comment, marked as answer]
You have either to create a derivate of action you have to extend, e. g. include `firewall*.conf` and add to `actionban` your custom code (basically too expensive)...
Or (much simpler) just add your action as second action in the jail (or default) section:
```ini
[sshd]
...
action = %(action_)s
         curlAction[mynamespace="whatever"]
```

**Reply by SystemSpam** [reply]
Hello sebreas.

Thank You for Your quick reply.

It' works.

Shame on me, that I don't mind to set it up in this way.
